### PR TITLE
👷(helm) preprod

### PR DESCRIPTION
**Pull request by AntoLC**
## Purpose

Configure the pre-production environment: https://desk-preprod.beta.numerique.gouv.fr/

Add the deploy workflow, this workflow will deploy the application to the selected tag:
Ex (https://github.com/numerique-gouv/people/releases): 
- Create a release with the version that you want (ex: v0.1.1), set it on the branch/commit that you want to deploy
- It will generate and push the docker images on docker hub - ⚠️ wait that is done (see CI job)
- Remove the `preprod` tag if it is already existing
- Set a new `preprod` tag to the version you just created (v0.1.1) and the correct branch.

Et voila, your preproduction will target the v0.1.1

## Proposal

- [x] 🔧(backend) activate https on oidc redirection
- [x] 👷(helm) preprod configuration
- [x] 👷(CI) add deploy workflow
- [x] 💚(CI) remove trigger workflow on push tags

## Demo

I created a demo branch / PR:
https://github.com/numerique-gouv/people/pull/236

I created the version `v0.1.0-demo`, see https://github.com/numerique-gouv/people/releases/tag/v0.1.0-demo.
I created the tag preprod that target the branch `demo/preprod`, see https://github.com/numerique-gouv/people/releases/tag/preprod

To put to production, we will just have to create our tag `production` that target the branch that we want to release, probably `main`.

### CODE REVIEW

File: src/helm/env.d/preprod/values.desk.yaml.gotmpl [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        key: DJANGO_SECRET_KEY
+    DJANGO_SETTINGS_MODULE: people.settings
+    DJANGO_SUPERUSER_PASSWORD:
+      secretKeyRef:
+        name: backend
+        key: DJANGO_SUPERUSER_PASSWORD
+    DJANGO_EMAIL_HOST: "snap-mail.numerique.gouv.fr"
+    DJANGO_EMAIL_PORT: 465
+    DJANGO_EMAIL_USE_SSL: True
+    DJANGO_SILENCED_SYSTEM_CHECKS: security.W008,security.W004
```
Comment (lebaudantoine) [inline src/helm/env.d/preprod/values.desk.yaml.gotmpl:26]: I think that these are now useless with your previous commit. It's a bit far in my head, but I think it fixes all the security warnings we had. Never mind, I can remove them in another PR, we would need to remove them in all envs.

File: src/helm/env.d/preprod/values.desk.yaml.gotmpl [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        key: database
+    POSTGRES_PASSWORD:
+      secretKeyRef:
+        name: postgresql.postgres.libre.sh
+        key: password
+    REDIS_URL:
+      secretKeyRef:
+        name: redis.redis.libre.sh
+        key: url
+
```
Comment (lebaudantoine) [inline src/helm/env.d/preprod/values.desk.yaml.gotmpl:83]: note: should we add your create super user command when your PR is merged?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Deploy
+
+on:
+  push:
+    tags:
+      - 'preprod'
+      - 'production'
```
Comment (lebaudantoine) [inline .github/workflows/deploy.yml:7]: Out of curiosity, do we already need the `production` tag?

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  start-test-on-preprod:
+    needs:
+      - notify-argocd
+    runs-on: ubuntu-latest
+    if: startsWith(github.event.ref, 'refs/tags/preprod')
+    steps:
+      -
+        name: Debug
+        run: |
+          echo "Start test when preprod is ready"
```
Comment (lebaudantoine) [inline .github/workflows/deploy.yml:41]: I'm not quite clear on what this job is responsible for. Could you provide a bit of context?

**Review by lebaudantoine — APPROVED** [review]
Looks good to me, great work !!

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (AntoLC) [reply .github/workflows/deploy.yml:7]: We can put it in the production PR if you want.

File: src/helm/env.d/preprod/values.desk.yaml.gotmpl (hunk 1 above)
Comment (AntoLC) [reply src/helm/env.d/preprod/values.desk.yaml.gotmpl:26]: Yes ok, I let you see.

**Pull request merged**